add project sub-nav link to org, display org title above description

mcbouslog opened the pull request:
Closes #3968.

If project linked to organization with `listed: true`, adds project sub-nav link to organization, display organization title above project description.

- https://project-landing-org-links.pfe-preview.zooniverse.org/projects/markb-panoptes/nfnorgtest-moths
- (with announcement banner) https://project-landing-org-links.pfe-preview.zooniverse.org/projects/markb-panoptes/nfnorgtest-wedigflplants

# Review Checklist

- [x] Does it work in all major browsers: Firefox, Chrome, Edge, Safari?
- [x] Does it work on mobile?
- [x] Can you `rm -rf node_modules/ && npm install` and app works as expected?
- [x] Did you deploy a staging branch? - https://project-landing-org-links.pfe-preview.zooniverse.org/
- [ ] If the component is in coffeescript, is it converted to ES6? Is it free of eslint errors? Is the conversion its own commit?
- [x] Are the tests passing locally and on Travis?


## Optional

- [ ] Have you replaced any `ChangeListener` or `PromiseRenderer` components with code that updates component state?
- [ ] If changes are made to the classifier, does the dev classifier still work?
- [ ] Have you [resized and compressed](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/image-optimization) any image you've added?
- [ ] Have you added in [flow type annotations](https://flowtype.org/docs/type-annotations.html)?
- [ ] Have you followed the [Springer guidelines for commit messages](https://github.com/springernature/frontend-playbook/blob/master/git/git.md#commit-messages)?

beckyrother commented:
Thanks @mcbouslog ! Couple small things: 
- Please add 10px more space between the organization name and the project name
- Will the organization name link back to the organization page?

mcbouslog commented:
Thanks @beckyrother! 

- I added `0.8em` spacing btw organization display name and project description, comes out to around 11px.
- Oops, forgot! Added link back to organization.

beckyrother commented:
@mcbouslog thought of something as you were going through the demo – the organization label needs a hover state or something to indicate it's a link. can it change to  #007482 on hover?

Same with the link in the header – can that have the external link icon that we talked about with James?

mcbouslog commented:
- styled link above project description
- added external-link icon next to sub-nav link back to org
- for consistency added external-link icon next to external links sub-nav items

srallen commented on app/pages/project/index.cjsx:
```diff
@@ -97,14 +98,19 @@ ProjectPageController = React.createClass
           awaitBackground = apiClient.type('backgrounds').get(project.links.background.id).catch((error) => [])
 
+          if project.links.organization?
+            awaitOrganization = project.get('organization', { listed: true }).catch((error) => [])
```
The caching layer in the client won't be used if you use the resource to get the organization. So I think right now two API requests are happening for the org, once with the include on line 101 and once here. Since you need to include this query param, then I think this line should remain and the addition to the include for the projects request should be removed.

mcbouslog replied:
not sure I've implemented the most elegant solution, but appears to be working as intended, let me know what you think

mcbouslog commented:
Didn't realize an org request filtered by `listed` isn't a valid option, created https://github.com/zooniverse/Panoptes/issues/2514.

srallen commented:
@mcbouslog I'm assuming that this PR has to wait for that related Panoptes issue to be resolved. Is that correct?

mcbouslog commented:
yep that's correct @srallen

mcbouslog commented on app/pages/project/project-page.cjsx:
```diff
@@ -158,14 +159,30 @@ ProjectPage = React.createClass
-            label = <i className="fa fa-#{iconForLabel} fa-fw fa-2x"></i>
-          <a key={i} href={url} className="tabbed-content-tab #{if iconForLabel then 'social-icon' else ''}" target="#{@props.project.id}#{url}">{label}</a>}
+            linkText = <i className="fa fa-#{iconForLabel} fa-fw fa-2x"></i>
+          <a key={i} href={url} className="tabbed-content-tab #{if iconForLabel then 'social-icon' else ''}" target="#{@props.project.id}#{url}">
```
utilize classnames package

simoneduca commented on app/pages/project/project-page.cjsx:
```diff
@@ -158,14 +159,30 @@ ProjectPage = React.createClass
+
         {rearrangedLinks.map ({label, url}, i) =>
-          unless !!label
+          if !!label
```
Could this be done without adding another `if-else` block to the already messy navbar logic? At least extract it to its own function? FYI, I'm preparing a PR that extracts the navbar to its own component and refactors it.

mcbouslog replied:
I think so, what you're doing certainly sounds preferable. I didn't refactor the general links logic with this PR, so I'm sure what you're doing will be an improvement. 

I'll refactor this PR to add the link in the sub-nav to a project's organization, if applicable, to work with your changes.

simoneduca replied:
I'm already struggling with reconstructing the original logic. So I'll leave adding the external link icon to you, if that's ok.

mcbouslog replied:
sounds good, I'll add the external link icon with this PR

mcbouslog commented:
update:
- related Panoptes issue/PR resolved
- holding on project sub-nav refactor

mcbouslog commented:
Originally I included updating external project links with the external-link icon in this PR, but I'll create a separate PR for that now, as I think it'll allow some refactoring to the current `renderProjectLinks` in `project-navbar.jsx`.

srallen commented on app/pages/project/index.cjsx:
```diff
@@ -115,6 +116,13 @@ ProjectPageController = React.createClass
+              .catch((error) => [])
+              .then((response) => if response?.display_name then response else null)
+          else
+            awaitOrganization = null
```
How about `Promise.resolve(null)` to be consistent with promises being returned in the `Promise.all()` for all of the linked project resources?

mcbouslog replied:
yes good call, I think I did as intended, but not 100% sure, let me know

srallen commented on css/project-page.styl:
```diff
@@ -181,16 +181,30 @@
+    text-decoration: none
+    text-transform: uppercase
+
+    &:hover
```
Hover styles should also be used for `:focus` too for keyboard users.

beckyrother commented:
We're not able to plan for all scenarios, so it may be hard to read no matter what we try to do. Mark suggested using TEAL_MID instead of ZOONIVERSE_TEAL, which is a great idea and might help a little.

mcbouslog commented:
honestly not sure why I did the slightly darker ZOONIVERSE_TEAL originally, should have done the lighter TEAL_MID, consistent with button hover/focus state color, my bad!
